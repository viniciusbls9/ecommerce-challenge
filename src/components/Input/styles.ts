import styled, { css } from 'styled-components'

export const InputWrapper = styled.div`
  width: 100%;

  & + div {
    margin-top: 1rem;
  }
`

export const Input = styled.input`
  ${({ theme }) => css`
    width: 100%;
    height: ${theme.spacings.xxlarge};
    border: 1px solid ${theme.colors.black};
    padding: 1rem;
    border-radius: 6px;

    &:focus {
      outline: none;
      border: 2px solid ${theme.colors.primary};
    }
  `}
`

export const InputSelect = styled.select`
  ${({ theme }) => css`
    width: 100%;
    height: 4.5rem;
    border: 1px solid ${theme.colors.primary};
    padding: 1rem;
  `}
`

export const Options = styled.option`
  ${({ theme }) => css`
    border-bottom: 1px solid ${theme.colors.black};
    padding: 2rem;
  `}
`
