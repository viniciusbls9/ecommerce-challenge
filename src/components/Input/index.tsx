import React, { InputHTMLAttributes } from 'react'
import * as S from './styles'

export interface InputProps extends InputHTMLAttributes<HTMLInputElement> {
  type?: string
}

const Input: React.FC<InputProps> = ({ type, ...rest }: InputProps) => {
  return (
    <S.InputWrapper>
      <S.Input type={type} {...rest} />
    </S.InputWrapper>
  )
}

export default Input
