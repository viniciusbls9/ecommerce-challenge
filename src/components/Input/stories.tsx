import { Story, Meta } from '@storybook/react/types-6-0'
import Input, { InputProps } from '.'

export default {
  title: 'Input',
  component: Input,
  args: {
    type: 'text'
  }
} as Meta

export const Default: Story = (args) => <Input {...args} />

const props = ['menor pre√ßo']

export const WithTypeSelect: Story<InputProps> = () => (
  <Input type="select">{props}</Input>
)
