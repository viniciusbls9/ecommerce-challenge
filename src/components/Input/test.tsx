import { renderWithTheme } from 'utils/helpers'
import { screen } from '@testing-library/react'

import Input from '.'

const props = ['menor preço']

describe('<Input />', () => {
  it('should render the Input', () => {
    const { container } = renderWithTheme(<Input type="text" />)

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render the Input with select type', () => {
    renderWithTheme(<Input type="select">{props}</Input>)

    expect(screen.getByRole('option', { name: 'menor preço' }))
  })
})
