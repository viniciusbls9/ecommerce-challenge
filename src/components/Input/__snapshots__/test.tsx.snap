// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> should render the Input 1`] = `
.c0 {
  width: 100%;
}

.c0 + div {
  margin-top: 10px;
}

.c1 {
  width: 100%;
  height: 5.6rem;
  border: 1px solid #030517;
  padding: 10px;
  border-radius: 6px;
}

.c1:focus {
  outline: none;
  border: 2px solid #F231A5;
}

<div
  class="c0"
>
  <input
    class="c1"
  />
</div>
`
